pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -335,6 +377,15 @@ func (dswp *desiredStateOfWorldPopulator) deleteProcessedPod(
 	delete(dswp.pods.processedPods, podName)
 }
 
+// Removes the specified pod records from failedProcessedPods
+func (dswp *desiredStateOfWorldPopulator) deleteFailedProcessPod(
```

Comment:
Looking around, `deleteFailedProcessPod` is called in `processPodVolumes` when everything succeeds and in `findAndRemoveDeletedPods` when a pod is removed from DSW.

Is there a possibility that a pod is created, `processPodVolumes` fails (e.g. PVC is not bound yet), message gets logged (and thus stored in `failedProcessPods`), but the pod is not stored in DSW, because all its volumes failed in `processPodVolumes`? The stored message will be never deleted when the pod is deleted.

Reply:
Yeap, that could be, and the result is the map is growing bigger and bigger. So a solution for this I can imagine right now is we add another go-routine in DSW to check the records in the map, if the msg is old enough(longer than the failureEventInterval), let's delete it from the map. What do you think of this, @msau42 and @jsafrane ?